newrelic-agent/src/main/java/com/newrelic/agent/OutboundHeadersMap.java

```diff
@@ -8,26 +8,42 @@
 package com.newrelic.agent;
 
 import java.util.HashMap;
+import java.util.Map;
 
 import com.newrelic.api.agent.HeaderType;
 import com.newrelic.api.agent.OutboundHeaders;
 
-@SuppressWarnings("serial")
-public class OutboundHeadersMap extends HashMap<String, String> implements OutboundHeaders {
+/**
+ * An {@link OutboundHeaders} that is backed by a map.
+ */
+public class OutboundHeadersMap implements OutboundHeaders {
```

Comment:
@jack-berg  Did you remove the extends HashMap and add returning a copy of the internal map because it's more idiomatic or was this necessary for the API to work?    I'm trying to piece together why `CrossProcessTransactionStateImpl` needed to change and this is the only clue I see so far

Reply:
Yup we should favor composition over inheritance. Just a bit of improvement while I was in the neighborhood.